src/runtime/button-api.js

```diff
@@ -122,4 +121,51 @@ export class ButtonApi {
     button.addEventListener('click', callback);
     return button;
   }
+
+  /**
+   *
+   * @param {!../api/subscriptions.ButtonOptions|function():null} optionsOrCallback
+   * @return {!../api/subscriptions.ButtonOptions}
+   * @private
+   */
+  getOptions_(optionsOrCallback) {
+    const options = /** @type {?../api/subscriptions.ButtonOptions} */
+        (typeof optionsOrCallback != 'function' ?
+        optionsOrCallback : null);
+    return options || {'theme': Theme.LIGHT, 'lang': 'en'};
+  }
+
+  /**
+   *
+   * @param {?../api/subscriptions.ButtonOptions|function():null} optionsOrCallback
+   * @param {function():null=} opt_callback
+   * @return {function():null|function(Event):boolean}
+   * @private
+   */
+  getCallback_(optionsOrCallback, opt_callback) {
+    const callback = /** @type {function():null|function(Event):boolean} */ (
+      (typeof optionsOrCallback == 'function' ? optionsOrCallback : null) ||
+          opt_callback);
+    return callback;
+  }
+
+  /**
+   * @param {!./deps.DepsDef} deps
+   * @param {!Element} button
+   * @param {!../api/subscriptions.ButtonOptions|function()} optionsOrCallback
+   * @param {function():null=} opt_callback
+   * @return {!Element}
+   */
+  attachSmartButton(deps, button, optionsOrCallback, opt_callback) {
+    const options = this.getOptions_(optionsOrCallback);
+    const callback = /** @type {function():null} */
+        (this.getCallback_(optionsOrCallback, opt_callback));
+    let theme = options && options['theme'];
```

Comment:
The `theme` never used?

Reply:
Re-factored this, moved to `getOptions_`

Reply:
removed now.